### DISCUSSION TITLE
Is libomni_vlm open source?

### DISCUSSION THREAD

**Discussion by pinyin** [NexaAI/nexa-sdk#295] [Q&A] [answered]
Hi, I noticed that the inference logic of omnivision (libomni_vlm_shared) seems to be downloaded during install.
If that's correct, may I ask if it is open-sourced yet? Thanks.

**Comment by zhiyuan8** [comment, marked as answer]
@pinyin libomni_vlm is generated during compilation of the project, you can find more from our code implementations here:
https://github.com/NexaAI/nexa-sdk/blob/main/nexa/gguf/nexa_inference_vlm_omni.py
https://github.com/NexaAI/nexa-sdk/blob/main/nexa/gguf/llama/omni_vlm_cpp.py

**Reply by pinyin** [reply]
> @pinyin libomni_vlm is generated during compilation of the project, you can find more from our code implementations here: https://github.com/NexaAI/nexa-sdk/blob/main/nexa/gguf/nexa_inference_vlm_omni.py https://github.com/NexaAI/nexa-sdk/blob/main/nexa/gguf/llama/omni_vlm_cpp.py

Hi, thanks for the reply. I've noticed these two files before, may I ask where are the codes that the dynamic lib is compiled from? I'm trying to understand whether I can integrate omnivision into my project, thanks for your great work.
